From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EL10 Compatibility <el10@compat.local>
Date: Wed, 19 Mar 2025 00:00:00 +0000
Subject: [PATCH] el10: remove GnomeQR eager import - typelib not in EL10

GnomeQR is from an unreleased gnome-desktop branch and is not packaged
for EL10 or EPEL10. The eager version-check import in dependencies.js
causes gnome-shell to crash at startup with:

  JS ERROR: Error: Requiring GnomeQR, version 4.0: Typelib file for
  namespace 'GnomeQR', version '4.0' not found

The qrCode.js module is only loaded during active web-login sessions
(imported by webLogin.js which is only triggered by an IDP redirect),
so removing the eager startup check is sufficient. Users will not see
a QR code in the web-login dialog, but gnome-shell will start normally.
---
 js/misc/dependencies.js | 1 -
 1 file changed, 0 insertions(+), 1 deletion(-)

diff --git a/js/misc/dependencies.js b/js/misc/dependencies.js
index 0000000..0000001 100644
--- a/js/misc/dependencies.js
+++ b/js/misc/dependencies.js
@@ -16,7 +16,6 @@ import 'gi://GioUnix?version=2.0';
 import 'gi://GDesktopEnums?version=3.0';
 import 'gi://GdkPixbuf?version=2.0';
 import 'gi://GnomeBG?version=4.0';
-import 'gi://GnomeQR?version=4.0';
 import 'gi://GnomeDesktop?version=4.0';
 import 'gi://Graphene?version=1.0';
 import 'gi://GUdev?version=1.0';
--
2.48.1
